src/Service/ZGWDocumentToFileService.php

```diff
@@ -0,0 +1,385 @@
+            $fileParts[] = $this->createFilePart($objectEntity, $iterator, ceil(($data['bestandsomvang'] / $parts)), $objectEntity->getLock());
+        }
+
+        $this->entityManager->persist($file);
+
+        if ($this->data['method'] === 'POST') {
+            $objectEntity->hydrate(['bestandsdelen' => $fileParts, 'lock' => $lock, 'locked' => true, 'inhoud' => $this->generateDownloadEndpoint($objectEntity->getId()->toString(), $downloadEndpoint)]);
+        }
+
+        if ($this->data['method'] !== 'POST') {
+            $objectEntity->hydrate(['bestandsdelen' => $fileParts, 'inhoud' => $this->generateDownloadEndpoint($objectEntity->getId()->toString(), $downloadEndpoint)]);
+        }
+
+        $this->entityManager->persist($objectEntity);
+        $this->entityManager->flush();
+
+        if ($setResponse === true) {
+            $this->data['response'] = new Response(
+                \Safe\json_encode($objectEntity->toArray()),
+                $this->data['method'] === 'POST' ? 201 : 200,
+                ['content-type' => 'application/json']
+            );
+        }
+
+    }//end getFilePart()
+
+
+    /**
+     * Creates or updates a file associated with a given ObjectEntity instance.
+     *
+     * This method handles the logic for creating or updating a file based on
+     * provided data. If an existing file is associated with the ObjectEntity,
+     * it updates the file's properties; otherwise, it creates a new file.
+     * It also sets the response data based on the method used (POST or other)
+     * and if the `$setResponse` parameter is set to `true`.
+     *
+     * @param ObjectEntity $objectEntity     The object entity associated with the file.
+     * @param array        $data             Data associated with the file such as title, format, and content.
+     * @param Endpoint     $downloadEndpoint Endpoint to use for downloading the file.
+     * @param bool         $setResponse      Determines if a response should be set, default is `true`.
+     *
+     * @return void
+     */
+    public function createOrUpdateFile(ObjectEntity $objectEntity, array $data, Endpoint $downloadEndpoint, bool $setResponse=true): void
+    {
+        if ($objectEntity->getValueObject('inhoud') !== false && $objectEntity->getValueObject('inhoud')->getFiles()->count() > 0) {
+            // Get the file from the inhoud object.
+            $file = $objectEntity->getValueObject('inhoud')->getFiles()->first();
+        }
+
+        if ($objectEntity->getValueObject('inhoud') !== false && $objectEntity->getValueObject('inhoud')->getFiles()->count() === 0) {
+            // Create the file with the data.
+            $file = $this->createFile($objectEntity, $data);
+        }
+
+        if ($data['inhoud'] !== null && $data['inhoud'] !== '' && filter_var($data['inhoud'], FILTER_VALIDATE_URL) === false) {
+            $file->setSize(mb_strlen(base64_decode($data['inhoud'])));
```

Comment:
f.y.i.
Validation for this checks size in bytes:
![image](https://github.com/CommonGateway/FirstRegistrationBundle/assets/3785488/fb9c4531-ad1d-4f18-b124-aa9417a538ae)
Maybe we should set size in bytes

Reply:
As discussed in slack: that is exactly what we do, so closing this comment